### PR TITLE
Test Coverage for #672

### PR DESCRIPTION
**Pull request by jorg3lopez**
# Test Coverage

This PR adds two more unit tests to cover the test coverage .

## Issue

#672

### CODE REVIEW

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestratorTest.groovy [hunk 1]
```diff
@@ -222,6 +222,21 @@ class PartnerMetadataOrchestratorTest extends Specification {
         1 * mockPartnerMetadataStorage.readMetadata(receivedSubmissionId) >> Optional.of(metadata)
     }
 
+    def "getMetadata retrieves metadata successfully when receiver is present and sentSubmissionId is missing"() {
+        given:
+        String receivedSubmissionId = "receivedSubmissionId"
+        def bearerToken = "token"
+        def metadata = new PartnerMetadata(receivedSubmissionId, null, "sender", "receiver", Instant.now(), "hash")
```
Comment (jorg3lopez) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestratorTest.groovy:229]: What is the probability of the sentSubmissionId being null?

**Comment by jorg3lopez** [comment]
Just noticed we need to add test coverage for DatabasePartnerMetadataStorage so I will put it back to DRAFT mode

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestratorTest.groovy (hunk 1 above)
Comment (basiliskus) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestratorTest.groovy:229]: It's possible if `sendOrder` fails in `SendOrderUseCase.convertAndSend`. But it's not possible for `sentSubmissionId` to be null and `receiver` to be non null given `sentSubmissionId` is needed to get the `receiver`

File: shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/utils/SyncRetryTaskTest.groovy [hunk 2]
```diff
@@ -43,4 +43,28 @@ class SyncRetryTaskTest extends Specification {
         def exception = thrown(RetryFailedException)
         exception.getCause().getClass() == Exception
     }
+
+    def "should handle thread interruption"() {
```
Comment (basiliskus) [inline shared/src/test/groovy/gov/hhs/cdc/trustedintermediary/utils/SyncRetryTaskTest.groovy:47]: Thank you for adding this test!

**Review by jcrichlake — APPROVED** [review]

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestratorTest.groovy (hunk 1 above)
Comment (basiliskus) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataOrchestratorTest.groovy:235]: Can you change this test to be `getMetadata retrieves metadata successfully when sentSubmissionId and receiver are missing`, which will be more realistic scenario?

**Pull request merged**